src/main/java/io/fabric8/elasticsearch/plugin/KibanaUserReindexAction.java

```diff
@@ -78,69 +82,71 @@ public void apply(String action, ActionRequest request,
 	public void apply(String action, ActionResponse response,
 			ActionListener listener, ActionFilterChain chain) {
 		
-		logger.debug("Response with Action '{}' and class '{}'", action, response.getClass());
-		
-		if ( containsKibanaUserIndex(response) ) {
-		
-			if ( response instanceof IndexResponse ) {
-				final IndexResponse ir = (IndexResponse) response;
-				String index = getIndex(ir);
-				
-				response = new IndexResponse(index, ir.getType(), ir.getId(), ir.getVersion(), ir.isCreated());
-			}
-			else if ( response instanceof GetResponse ) {
-				response = new GetResponse(buildNewResult((GetResponse) response));
-			}
-			else if ( response instanceof DeleteResponse ) {
-				final DeleteResponse dr = (DeleteResponse) response;
-				String index = getIndex(dr);
-				
-				response = new DeleteResponse(index, dr.getType(), dr.getId(), dr.getVersion(), dr.isFound());
-			}
-			else if ( response instanceof MultiGetResponse ) {
-				final MultiGetResponse mgr = (MultiGetResponse) response;
-				
-				MultiGetItemResponse[] responses = new MultiGetItemResponse[mgr.getResponses().length];
-				int index = 0;
-				
-				for ( MultiGetItemResponse item : mgr.getResponses() ) {
-					GetResponse itemResponse = item.getResponse();
-					Failure itemFailure = item.getFailure();
+		if ( enabled ) {
```

Comment:
We're doing this here because the module where we add this Action doesn't have the settings injected so unfortunately we can't prevent this from being added then.